jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -133,9 +134,51 @@ public void assignGroupToRole(final String serverTemplateId, final String contai
     }
 
     @Override
-    public void assignGroupAndUserToRole(final String serverTemplateId, final String containerId, final String caseId, final String roleName, final String user, final String group) {
-        client.assignGroupToRole(containerId, caseId, roleName, group);
-        client.assignUserToRole(containerId, caseId, roleName, user);
+    public void assignToRole(final String serverTemplateId, final String containerId, final String caseId,
+                             final String roleName, final List<String> usersList, final List<String> groupsList) {
+        CaseRoleAssignmentSummary ras = getCaseInstance(serverTemplateId, containerId, caseId).getRoleAssignments()
+                .stream()
+                .filter(ra -> roleName.equals(ra.getName()))
+                .findFirst().orElse(null);
```

Comment:
This looks very similar to what **getRoleAssignment** method defined in the presenter is doing.
Can this logic be delegated to presenter class and avoid fetching the case instance again from kie server?
That _hasAssignment_ is practically the result of this calculation, no?